lib/Graph/Context.cpp

```diff
@@ -46,6 +46,13 @@ void Context::clear() {
   map_.clear();
 }
 
+void Context::clone(Context &cloned) const {
+  cloned.clear();
+  for (auto PH : map_) {
+    cloned.insert(PH.first, PH.second->clone());
```

Comment:
imo it's hard to read without knowing what's in PH.first/second.

Reply:
Does my change to pull out the types of PH make it easier to parse?